### PR TITLE
Accept uuid types Iss30

### LINKED ISSUE #30: unrecognized PostgreSQL field type uuid (id:2950)
Connecting to a table with field type uuid generates a warning.

###  Table creation 
```SQL
CREATE TABLE prd_tv_sandbox.r_gender
(
respid UUID,
wave_id UUID,
gender INTEGER,
PRIMARY KEY (respid,wave_id)
);
```

## # The R:
```r
library (RPostgres)
library ( dplyr)
library ( DBI )


con_sb <- src_postgres(dbname = "the_db",   user = "[id]", password = "***")

gen_db <-  tbl ( con_sb, "r_gender" )
```

Warning messages:
1: In postgresqlExecStatement(conn, statement, ...) :
  RS-DBI driver warning: (unrecognized PostgreSQL field type uuid (id:2950) in column 0)
2: In postgresqlExecStatement(conn, statement, ...) :
  RS-DBI driver warning: (unrecognized PostgreSQL field type uuid (id:2950) in column 1)
```

### PR DESCRIPTION
**Pull request by JackStat**
If everyone is ok with uuid types being character classes in R then this is ready to merge.

Fixes #30.

### CODE REVIEW

File: tests/testthat/test-tables.R [hunk 1]
```diff
@@ -0,0 +1,55 @@
+context("Convenience Functions")
```
Comment (krlmlr) [inline tests/testthat/test-tables.R]: The generic functions are now mostly (if not entirely) tested by the DBitest package. I'd be happy to review pull requests for new tests there.

File: tests/testthat/test-dbGetQuery.R [hunk 2]
```diff
@@ -24,3 +24,22 @@ test_that("JSONB format is recognized", {
...
+
+  dbGetQuery(con, "CREATE TEMPORARY TABLE fuutab
+    (
+    fuu UUID,
+    name VARCHAR(255) NOT NULL
+    );")
+
+  dbGetQuery(con, "INSERT INTO fuutab(fuu, name) values ('c44352c0-72bd-11e5-a7f3-0002a5d5c51b', 'bob');")
+
+  expect_that(dbGetQuery(con, 'SELECT * FROM fuutab'), not(gives_warning()))
```
Comment (krlmlr) [inline tests/testthat/test-dbGetQuery.R]: Would you mind adding an expectation about the actual value returned when querying that table?

File: tests/testthat/test-dbGetQuery.R [hunk 3]
```diff
@@ -1,6 +1,6 @@
 context("dbGetQuery")
 
-test_that("special charaters work", {
+test_that("special characters work", {
```
Comment (krlmlr) [inline tests/testthat/test-dbGetQuery.R]: This test is removed in #56; could you please undo?

**Comment by JackStat** [comment]
@krlmlr Thanks for the feedback. I went through and made those modifications.

File: src/PqResult.h [hunk 4]
```diff
@@ -280,6 +280,8 @@ class PqResult : boost::noncopyable {
       case 1184: // TIMESTAMPTZOID
       case 1186: // INTERVAL
       case 1266: // TIMETZOID
+      case 3802: // JSONB
```
Comment (krlmlr) [inline src/PqResult.h]: This change seems unrelated to the original issue. I would be better to open a new PR for this, but I don't mind including it if you could provide a test.

File: tests/testthat/test-dbGetQuery.R [hunk 5]
```diff
@@ -24,3 +24,25 @@ test_that("JSONB format is recognized", {
...
+    fuu UUID,
+    name VARCHAR(255) NOT NULL
+    );")
+
+  uuid <- 'c44352c0-72bd-11e5-a7f3-0002a5d5c51b'
+
+  dbGetQuery(con, paste0("INSERT INTO fuutab(fuu, name) values ('", uuid, "', 'bob');"))
+
+  expect_that(dbGetQuery(con, 'SELECT * FROM fuutab'), not(gives_warning()))
+  expect_equal(dbGetQuery(con, 'SELECT * FROM fuutab')$fuu, uuid)
```
Comment (krlmlr) [inline tests/testthat/test-dbGetQuery.R]: New style, and also can do it with one query only:

```
expect_warning(
  expect_equal(...),
  NA
)
```

File: tests/testthat/test-dbGetQuery.R [hunk 6]
```diff
@@ -24,3 +24,25 @@ test_that("JSONB format is recognized", {
   expect_that(dbGetQuery(con, 'SELECT * FROM test2'), not(gives_warning()))
 
 })
+
+
+test_that("uuid format is recognized", {
+
+  con <- dbConnect(RPostgres::Postgres())
```
Comment (krlmlr) [inline tests/testthat/test-dbGetQuery.R]: `con` needs to be closed: `on.exit(..., add = TRUE)`

Comment (JackStat) [inline tests/testthat/test-dbGetQuery.R]: @krlmlr What file would you place on.exit?  Usually I have used this in each test
```
dbDisconnect(con)
```
but that may not be the most efficient way

Comment (krlmlr) [inline tests/testthat/test-dbGetQuery.R]: It's fine to open a new connection for each test (and closes it). To me, test isolation is more important than the small performance overhead.

**Comment by krlmlr** [comment]
Thanks! The dbDisconnect() should better be in on.exit(), but this doesn't matter too much for the tests.

**Comment by JackStat** [comment]
@krlmlr Thank you for the feedback and help. I really appreciate it.
